ComponentKitTests/Scope/CKComponentScopeTests.mm

```diff
@@ -219,4 +219,28 @@ - (void)testComponentScopeHandleGlobalIdentifierIsTheSameBetweenSiblingsWithComp
   }
 }
 
+#pragma mark - Component scope override
+
+- (void)testComponentScopeCanBeOverridden
+{
+  CKComponentScopeRoot *root = [CKComponentScopeRoot rootWithListener:nil];
+  CKThreadLocalComponentScope threadScope(root, {});
+  CKThreadLocalComponentScope *threadScopePtr = &threadScope;
+
+  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
@adamjernst That would work but @javache brings up a good point that it would be a bit more complex.

Let's try the test out as-is and go with the more complicated option if it proves to be flaky.